ietf/templates/meeting/session_details_panel.html

```diff
@@ -320,51 +320,58 @@ <h3 class="mt-4">Notes and recordings</h3>
-                        {% for r in recordings %}
-                            {% if r.get_href and 'youtu' in r.get_href %}
-                                <tr>
-                                    <td>
-                                        <a href="{{ r.get_href }}"><i class="bi bi-file-slides"></i> {{ r.title }}</a>
-                                    </td>
-                                </tr>
-                            {% endif %}
-                        {% endfor %}
-                        {# Finally, any other recordings #}
-                        {% for r in recordings %}
-                            {% if r.get_href and not 'audio' in r.get_href and not 'youtu' in r.get_href %}
-                                <tr>
-                                    <td>
-                                        <a href="{{ r.get_href }}"><i class="bi bi-file-play"></i> {{ r.title }}</a>
-                                    </td>
-                                </tr>
-                            {% endif %}
-                        {% endfor %}
-                    {% endif %}
-                {% endwith %}
-                {% if session.video_stream_url %}
+            {% with session.recordings as recordings %}
+                {% if recordings %}
+                    {# There's no guaranteed order, so this is a bit messy: #}
+                    {# First, the audio recordings, if any #}
+                    {% for r in recordings %}
+                        {% if r.get_href and 'audio' in r.get_href %}
+                            <tr>
+                                <td>
+                                    <a href="{{ r.get_href }}"><i class="bi bi-file-play"></i> {{ r.title }}</a>
+                                </td>
+                            </tr>
+                        {% endif %}
+                    {% endfor %}
+                    {# Then the youtube recordings #}
+                    {% for r in recordings %}
+                        {% if r.get_href and 'youtu' in r.get_href %}
+                            <tr>
+                                <td>
+                                    <a href="{{ r.get_href }}"><i class="bi bi-file-slides"></i> {{ r.title }}</a>
+                                </td>
+                            </tr>
+                        {% endif %}
+                    {% endfor %}
+                    {# Finally, any other recordings #}
+                    {% for r in recordings %}
+                        {% if r.get_href and not 'audio' in r.get_href and not 'youtu' in r.get_href %}
+                            <tr>
+                                <td>
+                                    <a href="{{ r.get_href }}"><i class="bi bi-file-play"></i> {{ r.title }}</a>
+                                </td>
+                            </tr>
+                        {% endif %}
+                    {% endfor %}
+                {% endif %}
+                {% if session.session_recording_url %}
```

Comment:
Move this out of the with.

Putting the meetecho recording name directly on session is very new, and stands independent of whether there are any documents of type recording associated with the session.